File: backend/app/api/interview.py
from fastapi import APIRouter

from app.ai.ollama_service import ollama
from app.prompts.question_prompt import build_question_prompt

from app.schemas.interview import (
    QuestionRequest,
    QuestionResponse,
)

router = APIRouter(
    prefix="/interview",
    tags=["Interview"],
)


@router.post(
    "/question",
    response_model=QuestionResponse,
)
def generate_question(
    request: QuestionRequest,
):
    prompt = build_question_prompt(
        request.topic,
        request.difficulty,
    )

    question = ollama.generate(prompt)
    print(question)
    print(type(question))

    return QuestionResponse(
        topic=request.topic,
        difficulty=request.difficulty,
        question=question,
    )

File: frontend/src/api/interview.ts
import api from "./axios";

import type {
  EvaluationRequest,
  EvaluationResponse,
  QuestionRequest,
  QuestionResponse,
} from "../types/interview";

export async function generateQuestion(
  request: QuestionRequest
): Promise<QuestionResponse> {
  const response = await api.post(
    "/interview/question",
    request
  );

  return response.data;
}

export async function evaluateAnswer(
  request: EvaluationRequest
): Promise<EvaluationResponse> {
  const response = await api.post(
    "/interview/evaluate",
    request
  );

  return response.data;
}